src/adm/settings.rs

```diff
@@ -37,7 +40,12 @@ impl From<&Settings> for AdmSettings {
         if settings.adm_settings.is_empty() {
             return Self::default();
         }
-        serde_json::from_str(&settings.adm_settings).expect("Invalid ADM Settings")
+        if Path::new(&settings.adm_settings).exists() {
+            if let Ok(f) = File::open(&settings.adm_settings) {
+                return serde_json::from_reader(f).expect("Invalid ADM Settings file");
+            }
```

Comment:
Do you think we should return an error if the file open fails? Otherwise we'll try to read it as a JSON string even though the path does in fact exist (since we passed the check on L43). Might not be worth it, but thought I'd mention it 🤷🏻

Reply:
Not sure I follow.

If `adm_settings` is empty, we use the default.

If `adm_settings` resolves to a valid file path, then we read the content of that file as the `adm_settings`.

If it's not a valid path (because it's a JSON blob), try reading that blob as JSON and use that as `adm_settings`. 

Otherwise give up with a `panic!`

Reply:
I'll add a comment that makes that clearer. Thanks!

Reply:
Ah I see, I was talking about the case where the file path is valid but `File::open` returns an `Err`, but I guess that's pretty unlikely since we're checking `Path::exists` first